### ISSUE TITLE
Question: get trial id by exporting

### ISSUE THREAD

**Issue by Annika10** [microsoft/nni#4447]
I have just a small question by running the [multi_trial example](https://github.com/microsoft/nni/blob/v2.5/examples/nas/oneshot/spos/multi_trial.py). Is it possible to get the trial ID when exporting the model dict at the end? Would be nice to get a direct connection which trial on the Web UI is the best model without checking all mutations. 

It would be even nicer if the lightning logs are also named like the trial ID for better comparing with tensorboard. I know I can use tensorboard with the lightning logs produced by the trainer, but the trials are not named with the trial ID. On the other hand, when I want to use tensorboard via the Web UI it doesn't work because the logs are not saved in the trials folder of the nni-experiments. I tried solving this by defining the log directory like explained [here](https://nni.readthedocs.io/en/stable/Tutorial/Tensorboard.html?highlight=tensorboard#save-logs) but this [doesn't work](https://github.com/microsoft/nni/issues/3028) when not using the `nnictl`command.

**Comment by ultmaster** [comment]
These are two feature reqeusts.

For the first one, I think @QuanluZhang can help with that.
For the second one, I'm thinking that we can set `log_dir` automatically in evaluators like Classification. Do you prefer them saved into `lightning_logs` or `nni-experiments`?

**Comment by Annika10** [comment]
Yeah right, I want to find the best model with the WebUI, or maybe even the k best models. 

>  Otherwise, it can be simply accomplished if you looked at the implementation of `export_top_models`.

But how? I can't see where I can get the trial id of the exported model. 

> For the second one, I'm thinking that we can set `log_dir` automatically in evaluators like Classification. Do you prefer them saved into `lightning_logs` or `nni-experiments`?

I would prefer having the logs in `lightning_logs` with their corresponding trial ID

**Comment by ultmaster** [comment]
> I would prefer having the logs in lightning_logs with their corresponding trial ID

Got it. I'll consider doing that.

**Comment by heibaidaolx123** [comment]
I have the same question, when calling export_top_models, I need the corresponding trial IDs.
Is there a way to do that? @ultmaster 
Thanks.

**Comment by ultmaster** [comment]
Unfortunately as of 2023 we don't currently have bandwidth to implement this feature. Community contribution is welcomed.

**Comment by heibaidaolx123** [comment]
> Unfortunately as of 2023 we don't currently have bandwidth to implement this feature. Community contribution is welcomed.

OK. Got it. 
BTW, is there a way to get the mutation of the space in a trial during NAS? If yes, I can keep the <trialID, mutation, score> in my code, and export the models with its trialID.
Thanks. @ultmaster

**Comment by heibaidaolx123** [comment]
Will try.
Thanks for your patience and detailed explanation. 
@ultmaster

**Comment by heibaidaolx123** [comment]
@ultmaster Sorry to bother again.
I tried nni.get_current_parameter() in trials, the returned dict contained the mutaion. 
Is it a correct method?

**Comment by ultmaster** [comment]
You can do this. But nni.get_current_parameter is only available within trials.
